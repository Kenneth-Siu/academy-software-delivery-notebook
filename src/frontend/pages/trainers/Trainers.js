import React, { useEffect, useState } from "react";
import { getTrainers } from "../../api/trainerApi";
import "./Trainers.scss";

export default function Trainers() {
    const [trainers, setTrainers] = useState(null);

    useEffect(async () => {
        setTrainers(await getTrainers());
    }, []);

    return (
        <main className="trainers-page">
            <h1>Trainers</h1>
            {trainers ? (
                <table>
                    <tbody>
                        {trainers.map((trainer) => (
                            <tr key={trainer.id}>
                                <td>{trainer.name}</td>
                            </tr>
                        ))}
                    </tbody>
                </table>
            ) : (
                <p>Loading...</p>
            )}
        </main>
    );
}
