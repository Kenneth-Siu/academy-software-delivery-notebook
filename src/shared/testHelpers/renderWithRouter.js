import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

export default function renderWithRouter(component) {
    return render(<MemoryRouter>{component}</MemoryRouter>);
}
