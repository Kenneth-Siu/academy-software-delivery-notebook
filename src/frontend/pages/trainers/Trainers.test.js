import React from "react";
import { act, waitFor } from "@testing-library/react";
import renderWithRouter from "../../../shared/testHelpers/renderWithRouter";
import Trainers from "./Trainers";

let fetchMock;

beforeAll(() => {
    fetchMock = jest.fn(async () => {
        return {
            ok: true,
            json: () => [
                { id: 1, name: "Apple pie" },
                { id: 4, name: "Halloumi Hamburgers" },
                { id: 2, name: "Tomato Soup" },
            ],
        };
    });
    window.fetch = fetchMock;
});

afterAll(() => {
    delete window.fetch;
});

describe("Trainers page", () => {
    test("renders", async () => {
        // Arrange
        let tree;

        // Act
        act(() => {
            tree = renderWithRouter(<Trainers />);
        });
        await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1));

        // Assert
        expect(tree.container).toMatchSnapshot();
    });
});
