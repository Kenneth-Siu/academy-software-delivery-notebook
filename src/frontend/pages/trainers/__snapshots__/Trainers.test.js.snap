// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trainers page renders 1`] = `
<div>
  <main
    class="trainers-page"
  >
    <h1>
      Trainers
    </h1>
    <table>
      <tbody>
        <tr>
          <td>
            Apple pie
          </td>
        </tr>
        <tr>
          <td>
            Halloumi Hamburgers
          </td>
        </tr>
        <tr>
          <td>
            Tomato Soup
          </td>
        </tr>
      </tbody>
    </table>
  </main>
</div>
`;
